server/engine/engine.go

```diff
@@ -307,8 +313,10 @@ func (engine *Engine) runStep(step model.Step, execution *model.Execution, waitG
 			outputValues[v.ModuleName] = v.Values
 		}
 	}
-	engine.resolver.ResolveReferencesToOutputs(step.Parameters, outputValues)
-
+	err := engine.resolver.ResolveReferencesToOutputs(step.Parameters, outputValues)
+	if err != nil {
```

Comment:
If this code can't resolve outputs, the step that's about to be executed will fail. We can either stop it here or let it fail.
We need to test these situations to see what user experience with them is.

Reply:
I just did it the same way as we do just below when the StartDeployARMTemplate returns an error, which should be fine.